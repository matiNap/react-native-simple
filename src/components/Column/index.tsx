import React, { ReactNode } from 'react';
import { StyleSheet, View, StyleProp, ViewStyle } from 'react-native';
import { pickBackroundPaletteColor, getThemeColor } from '../../helpers';
import useTheme from '../ThemeProvider/useTheme';

interface Props {
    children: ReactNode[] | ReactNode;
    style?: StyleProp<ViewStyle>;
    borderRadius?: number;
    backgroundColor?: string;
    height?: string;
    justifyContent?:
        | 'space-between'
        | 'flex-start'
        | 'flex-end'
        | 'center'
        | 'space-around'
        | 'space-evenly'
        | undefined;
    align?:
        | 'flex-start'
        | 'flex-end'
        | 'center'
        | 'auto'
        | 'stretch'
        | 'baseline'
        | undefined;
}

export default function ({
    justifyContent,
    height,
    backgroundColor,
    style,
    children,
    align,
}: Props) {
    const theme = useTheme();
    return (
        <View
            style={[
                styles.container,
                {
                    justifyContent,
                    height,
                    backgroundColor: getThemeColor(
                        backgroundColor,
                        theme.currentPalette,
                        theme.currentPalette.primary,
                    ),
                    alignSelf: align,
                },
                style,
            ]}
        >
            {children}
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'column',
    },
});
