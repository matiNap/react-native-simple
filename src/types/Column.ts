import { StyleProp, ViewStyle } from 'react-native';
import { ReactNode } from 'react';

export default interface ColumnProps {
    children: ReactNode[] | ReactNode;
    style?: StyleProp<ViewStyle>;
    backgroundColor?: string;
    height?: string;
    justifyContent?:
        | 'space-between'
        | 'flex-start'
        | 'flex-end'
        | 'center'
        | 'space-around'
        | 'space-evenly'
        | undefined;
    align?:
        | 'flex-start'
        | 'flex-end'
        | 'center'
        | 'auto'
        | 'stretch'
        | 'baseline'
        | undefined;
}
