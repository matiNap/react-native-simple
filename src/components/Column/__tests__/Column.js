import React from 'react';
import { View } from 'react-native';
import { shallow } from 'enzyme';
import Column from '../index';
import toJson from 'enzyme-to-json';

describe('Column component', () => {
    it('Should render', () => {
        const wrapper = shallow(<Column />);

        expect(wrapper.length).toBe(1);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should render with style', () => {
        const wrapper = shallow(
            <Column
                style={{ backgroundColor: 'red' }}
                backgroundColor="blue"
                align="center"
            />,
        );

        expect(wrapper.length).toBe(1);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should render with childs', () => {
        const wrapper = shallow(
            <Column>
                <View />
                <View />
                <View />
                <View />
            </Column>,
        );

        expect(wrapper.length).toBe(1);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
