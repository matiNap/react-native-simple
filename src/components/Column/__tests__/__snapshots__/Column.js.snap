// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column component Should render 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "column",
      },
      Object {
        "alignSelf": undefined,
        "backgroundColor": undefined,
        "height": undefined,
        "justifyContent": undefined,
      },
      undefined,
    ]
  }
/>
`;

exports[`Column component Should render with childs 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "column",
      },
      Object {
        "alignSelf": undefined,
        "backgroundColor": undefined,
        "height": undefined,
        "justifyContent": undefined,
      },
      undefined,
    ]
  }
>
  <View />
  <View />
  <View />
  <View />
</View>
`;

exports[`Column component Should render with style 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "column",
      },
      Object {
        "alignSelf": "center",
        "backgroundColor": "blue",
        "height": undefined,
        "justifyContent": undefined,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
/>
`;
